src/utils/block-helper.ts

```diff
@@ -92,20 +92,12 @@ export function getPositionByKeyword(block: Block, fields: Array<string>) : Posi
       return getAttrSourcePosition(blockAttr);
     }
   }
-  
+
   return;
 }
 
-export function parse(filename: string, contents: string, cssBlocksOpts?: Options) {
-  const root = postcss.parse(contents);
-
+export function parse(filename: string, cssBlocksOpts?: Options) {
   const config = resolveConfiguration(cssBlocksOpts);
-
-  // Fetch block name from importer
-  let identifier = config.importer.identifier(null, filename, config);
-  let defaultName = config.importer.defaultName(identifier, config);
   let factory = new BlockFactory(config, postcss);
-
-  return factory.parse(root, filename, defaultName);
+  return factory.getBlockFromPath(filename);
```

Comment:
This method was doing basically the same thing as what `getBlockFromPath` does.

Reply:
wow, that's nice!